### ISSUE TITLE
Maximize window in windowed mode

### ISSUE THREAD

**Issue by maxhark** [iina/iina#4712]
**What you want IINA to do:**
I want an option to have the window take up the full screen without going full screen, like most other applications

**What IINA does currently:**
Currently, when trying to maximize the window by double clicking on the title bar, IINA maximizes as much as possible while preserving the ratio of the video. Since the menu bar at the top takes up space, this leads to it not covering the edges of the screen.
![Screenshot 2023-12-06 at 11 34 41 AM](https://github.com/iina/iina/assets/29798975/9370964e-594a-463e-8df1-c08b362c1793)

There should be an option to have it behave like most applications do, where the whole screen is covered. Black bars can be used to prevent the ratio of the video from being stretched.
![Screenshot 2023-12-06 at 11 33 01 AM](https://github.com/iina/iina/assets/29798975/c814bb9d-d482-45ff-b9cf-36bca359f15d)

Sometimes I want to watch content in "almost-fullscreen" I want to see nothing but the player menus and the content.

See the screenshot from VLC above.

**Comment by aronreisx** [comment]
I just started using Iina today and that was the first thing I tried to configure. All the modern player have this feature, kinda became a standard. Up votes for this one.

**Comment by TheLostTree** [comment]
I second this!

**Comment by svobs** [comment]
Implementing this feature requires rewriting most of IINA's window layout code because everything was written with the assumption that window size is the same as video size.

After months of effort I got this working (along with other layout features) but I ended up with several thousand lines of new code. Too much for a PR, realistically. So I recently published a [binary of my fork](https://github.com/svobs/iina-advance/releases/) if anyone would like to check it out and report any bugs. You can enable this feature by opening its `Settings` window and unchecking `UI` > `Window Geometry` > `Lock proportions to prevent empty margins around video`.

**Comment by ayurnero** [comment]
我也需要这个功能，这个功能还有个好处就是进度条也会被拉长，可以在鼠标点击进度条快进时更精细。

**Comment by ayurnero** [comment]
> 实现此功能需要重写 IINA 的大部分窗口布局代码，因为所有内容都是在假设窗口大小与视频大小相同的情况下编写的。
> 
> 经过几个月的努力，我得到了这个工作（以及其他布局功能），但最终我得到了几千行新代码。 实际上，对于公关来说太多了。 我最近发布了 [我的分支的二进制文件 ](https://github.com/svobs/iina-advance/releases/) 因此，如果有人想检查并报告任何错误， 。 您可以通过打开其来启用此功能 `Settings`窗口并取消选中 `UI`> `Window Geometry`> `Lock proportions to prevent empty margins around video`.

我下载了您的 1.0 版本，但是我进行任意自定义设置后再双击打开视频文件都会报错。

**Comment by ayurnero** [comment]
> Implementing this feature requires rewriting most of IINA's window layout code because everything was written with the assumption that window size is the same as video size.
> 
> After months of effort I got this working (along with other layout features) but I ended up with several thousand lines of new code. Too much for a PR, realistically. So I recently published a [binary of my fork](https://github.com/svobs/iina-advance/releases/) if anyone would like to check it out and report any bugs. You can enable this feature by opening its `Settings` window and unchecking `UI` > `Window Geometry` > `Lock proportions to prevent empty margins around video`.


I downloaded your 1.0 version, but after I made any custom settings and then double-clicked to open the video file, an error occurred.
After many tests, it was confirmed that the problem should be caused by setting it as the default player. After setting it as the default player, double-clicking the video file reported an error, showing that iina unexpectedly exited. There was no problem with starting iina first and then using other methods such as drag and drop to open the video file.
I don’t know if it’s related to the fact that the system is a 14.5 beta system.

**Comment by ayurnero** [comment]
> > Implementing this feature requires rewriting most of IINA's window layout code because everything was written with the assumption that window size is the same as video size.
> > After months of effort I got this working (along with other layout features) but I ended up with several thousand lines of new code. Too much for a PR, realistically. So I recently published a [binary of my fork](https://github.com/svobs/iina-advance/releases/) if anyone would like to check it out and report any bugs. You can enable this feature by opening its `Settings` window and unchecking `UI` > `Window Geometry` > `Lock proportions to prevent empty margins around video`.
> 
> I downloaded your 1.0 version, but after I made any custom settings and then double-clicked to open the video file, an error occurred. After many tests, it was confirmed that the problem should be caused by setting it as the default player. After setting it as the default player, double-clicking the video file reported an error, showing that iina unexpectedly exited. There was no problem with starting iina first and then using other methods such as drag and drop to open the video file. I don’t know if it’s related to the fact that the system is a 14.5 beta system.

This is still the case after the mac system is downgraded to 14.3. After setting iina as the default player, double-clicking to open a video file will crash.

**Comment by ayurnero** [comment]
å

> > > Implementing this feature requires rewriting most of IINA's window layout code because everything was written with the assumption that window size is the same as video size.
> > > After months of effort I got this working (along with other layout features) but I ended up with several thousand lines of new code. Too much for a PR, realistically. So I recently published a [binary of my fork](https://github.com/svobs/iina-advance/releases/) if anyone would like to check it out and report any bugs. You can enable this feature by opening its `Settings` window and unchecking `UI` > `Window Geometry` > `Lock proportions to prevent empty margins around video`.
> > 
> > 
> > I downloaded your 1.0 version, but after I made any custom settings and then double-clicked to open the video file, an error occurred. After many tests, it was confirmed that the problem should be caused by setting it as the default player. After setting it as the default player, double-clicking the video file reported an error, showing that iina unexpectedly exited. There was no problem with starting iina first and then using other methods such as drag and drop to open the video file. I don’t know if it’s related to the fact that the system is a 14.5 beta system.
> 
> This is still the case after the mac system is downgraded to 14.3. After setting iina as the default player, double-clicking to open a video file will crash.

The final confirmation is to uncheck "Always open files in a new window" and double-click to open the video file. iina will crash.

**Comment by svobs** [comment]
@ayurnero Just saw this. I'll take a look. I've been working on a 1.1 release already, but sounds like I may need to speed up my timeline.

I enabled issues for IINA Advance now. If you can, it would be better to [file an issue for that project](https://github.com/svobs/iina-advance/issues) directly. Thanks!

**Comment by ayurnero** [comment]
> @ayurnero Just saw this. I'll take a look. I've been working on a 1.1 release already, but sounds like I may need to speed up my timeline.
> 
> I enabled issues for IINA Advance now. If you can, it would be better to [file an issue for that project](https://github.com/svobs/iina-advance/issues) directly. Thanks!

Okay, thank you very much for your answer. I have re-submitted this question in the IINA Advance FAQ. I am looking forward to your 1.1 version. I think your version has a lot more customizable options, which is very good. I look forward to more of people like this version.

**Comment by abhishek-notes** [comment]
I agree, I wish this one gets done soon. It’s the only reason I’m leaving IINA, and the reason I stayed was the timeline preview frame, similar to YouTube. Thanks for your attention to this matter.

**Comment by abhishek-notes** [comment]
> I agree, I wish this one gets done soon. It’s the only reason I’m leaving IINA, and the reason I stayed was the timeline preview frame, similar to YouTube. Thanks for your attention to this matter.

BTW, for anybody else looking for this, they can try one from @svobs mentioned above, in his advanced version, it is working, i just realised. Thanks again for your attention, but i still wish it to be added in the default one.

**Comment by uiryuu** [comment]
Fixed by https://github.com/iina/iina/pull/6100
